internal/data/generated_slices.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by slice_generator_main.go. DO NOT EDIT.
+// To regenerate this file run `go run internal/data/internal/slice_generator_main.go`.
+
+package data
+
+import (
+	"sort"
+
+	otlpmetrics "github.com/open-telemetry/opentelemetry-proto/gen/go/metrics/v1"
+)
+
+// MetricsSlice logically represents a slice of Metric.
```

Comment:
Maybe also add a comment explaining that this is a reference type, if you pass by value
and callee modifies it the caller will see the modification.

Reply:
Done.